### PR TITLE
mbedit - address issue #1041

### PR DESCRIPTION
**Pull request by dwcaress**
Addressed TODO's left by Kurt in mbedit_prog.c in conjunction with issue #1041. The always true else statements have been commented out. Some variables converted to bool as suggested.  The logic questioned for the reverse cut filters is correct and has been left in place. When begin < end then the cut is in the interior of the swath. When begin > end then cuts are made at the outer edges of the swath.

### CODE REVIEW

**Comment by schwehr** [comment]
I think there will still be expressions that will always be true triggering warnings.  That will confuse people like me in the future as it is surprising to have an expression that isn't needed.  Maybe assert or a comment, but having the expressions in an if is misleading

**Comment by dwcaress** [comment]
I've dealt with the cases you identified and the other similar cases in mbedit_prog.c. What other expressions are you referring to?

File: src/mbedit/mbedit_prog.c [hunk 1]
```diff
@@ -3678,7 +3674,11 @@ int mbedit_open_file(char *file, int form, int savemode) {
 		}
 
 		/* handle esf edits */
-		const bool outputmode = output_mode != MBEDIT_OUTPUT_BROWSE;
+    int outputmode;
+    if (output_mode == MBEDIT_OUTPUT_BROWSE)
+      outputmode = MBP_ESF_NOWRITE;
+    else
+      outputmode = MBP_ESF_WRITE;
```
Comment (schwehr) [inline src/mbedit/mbedit_prog.c:3681]: Should do this if it's an int:

```c
const int outputmode = output_mode == MBEDIT_OUTPUT_BROWSE ? MBP_ESF_NOWRITE :  MBP_ESF_WRITE;
```

This is really confusing code.  `if (savemode || outputmode)` implies outputmode is a bool.

File: src/mbedit/mbedit_prog.c [hunk 2]
```diff
@@ -5337,10 +5336,9 @@ int mbedit_tsminmax(int iping, int nping, int data_id, double *tsmin, double *ts
 	if (iping >= 0 && nbuff > iping && nping > 0 && iping + nping - 1 < nbuff) {
 		mbedit_tsvalue(iping, data_id, tsmin);
 		*tsmax = *tsmin;
-		// TODO(schwehr): Localize value and value2?
-		double value;
-		double value2;
 		for (int i = iping; i < iping + nping; i++) {
+		  double value;
+		  double value2;
```
Comment (schwehr) [inline src/mbedit/mbedit_prog.c:5341]: value2 should then be pushed inside the `if (data_id == MBEDIT_PLOT_ROLL)`

**Comment by schwehr** [comment]
> I've dealt with the cases you identified and the other similar cases in mbedit_prog.c. What other expressions are you referring to?

I don't follow why the fix was done the way it was.

File: src/mbedit/mbedit_prog.c (hunk 1 above)
Comment (dwcaress) [reply src/mbedit/mbedit_prog.c:3681]: To me any use of a ternary operator is confusing  to read, but I have put it back mostly the way you want.

File: src/mbedit/mbedit_prog.c (hunk 2 above)
Comment (dwcaress) [reply src/mbedit/mbedit_prog.c:5341]: Done.

**Pull request merged**